### PR TITLE
Update contributing.md with how to contribute a config function

### PR DESCRIPTION
**Pull request by prachirp**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+
+## How to Contribute
+
+This catalog contains configuration functions to fetch, display, customize,
+update, validate, and apply Kubernetes configuration. Contribute functions
+to this catalog by making the following changes:
+
+1. Make the code changes implementing the function and any unit tests. The code
+   belongs under [functions/] in the appropriate directory.
+2. Create an example of how to run your function and put it under [examples/].
```
Comment (haiyanmeng) [inline CONTRIBUTING.md:19]: Currently, `examples` dir does not exist.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+3. Test running your function example imperatively and declaratively in [e2e tests]
+   These tests are run regularly and help catch regressions.
```
Comment (haiyanmeng) [inline CONTRIBUTING.md:21]: Add an example PR here

File: CONTRIBUTING.md (hunk 1 above)
Comment (prachirp) [reply CONTRIBUTING.md:19]: Created an examples directory with helm template example

File: CONTRIBUTING.md (hunk 2 above)
Comment (prachirp) [reply CONTRIBUTING.md:21]: Added PR showing helm template example and e2e tests

**Comment by prachirp** [comment]
cc @haiyanmeng @frankfarzan PTAL

**Comment by haiyanmeng** [comment]
LGTM

**Pull request merged**